src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/PackageLibs.targets

```diff
@@ -3,10 +3,47 @@
   <PropertyGroup>
     <PackagingTaskDir Condition="'$(PackagingTaskDir)' == ''">$(MSBuildThisFileDirectory)</PackagingTaskDir>
     <GenerationDefinitionFile Condition="'$(GenerationDefinitionFile)' == ''">$(MSBuildThisFileDirectory)Generations.json</GenerationDefinitionFile>
+    <RuntimeIdGraphDefinitionFile Condition="'$(RuntimeIdGraphDefinitionFile)' == '' AND Exists('$(ProjectDir)pkg/Microsoft.NETCore.Platforms/runtime.json')">$(ProjectDir)pkg/Microsoft.NETCore.Platforms/runtime.json</RuntimeIdGraphDefinitionFile>
+    <RuntimeIdGraphDefinitionFile Condition="'$(RuntimeIdGraphDefinitionFile)' == ''">$(MSBuildThisFileDirectory)runtime.json</RuntimeIdGraphDefinitionFile>
   </PropertyGroup>
 
   <UsingTask TaskName="ValidatePackageTargetFramework" AssemblyFile="$(PackagingTaskDir)Microsoft.DotNet.Build.Tasks.Packaging.dll"/>
+  <UsingTask TaskName="GetApplicableAssetsFromPackages" AssemblyFile="$(PackagingTaskDir)Microsoft.DotNet.Build.Tasks.Packaging.dll"/>
 
+  <Target Name="UpdatePkgProjProjectReferences" BeforeTargets="BeforeResolveReferences">
+    <!-- Update PkgProj references to call the GetValidationReport target -->
+    <ItemGroup>
+      <ProjectReference Condition="'%(Extension)'=='.pkgproj'">
+        <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+        <Targets>Build;GetPackageAssets</Targets>
+        <OutputItemType>PackageAsset</OutputItemType>
+      </ProjectReference>
+    </ItemGroup>
+  </Target>
+  
+  <PropertyGroup>
+    <!-- Resolve from pkgProjs before nuget packages -->
+    <ResolveNugetPackagesDependsOn>$(ResolveNugetPackagesDependsOn);ResolvePkgProjReferences</ResolveNugetPackagesDependsOn>
+    <ResolvePkgProjReferencesDependsOn>ResolveProjectReferences</ResolvePkgProjReferencesDependsOn>
+  </PropertyGroup>
+  
+  <Target Name="ResolvePkgProjReferences" Condition="'@(PackageAsset)' != ''" DependsOnTargets="$(ResolvePkgProjReferencesDependsOn)">
+    <PropertyGroup>
+      <!-- Temporary workaround -->
+      <NuGetTargetMoniker Condition="'$(NuGetTargetMoniker)' == '.NETPortable,Version=v4.5,Profile=Profile7'">DNXCore,Version=v5.0</NuGetTargetMoniker>
+    </PropertyGroup>
+    
+    <!-- Hack this for now to use TestNugetRuntimeId.  We probably want to run this in a way to get the compile assets for the
+         target platform and runtime assets for the RID-specific platform.  -->
+    <GetApplicableAssetsFromPackages PackageAssets="@(PackageAsset)" 
+                                     TargetMoniker="$(NuGetTargetMoniker)"
+                                     RuntimeTargetMoniker="$(TestNuGetTargetMoniker)"
+                                     TargetRuntime="$(TestNugetRuntimeId)"
```

Comment:
@weshaggard  I don't think we need to introduce TestRID. I was thinking to use TestNugetRuntimeId and this will still be derived from TestTFM. do we really need TestRID property?